enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,550 @@
+    Input("target_year","value"),
+    Input("target_season","value"),
+)
+def write_map_title(target_year,target_season):
+    season_str = select_season(target_season)
+    map_title = ("Climate suitability for " + season_str + " season in " + str(target_year))
+
+    return map_title
+
+@FLASK.route(f"{TILE_PFX}/<int:tz>/<int:tx>/<int:ty>")
+def cropSuit_layers(tz, tx, ty):
+    parse_arg = pingrid.parse_arg
+    data_choice = parse_arg("data_choice")
+    target_season = parse_arg("target_season")
+    target_year = parse_arg("target_year", float)  
+    data_choice = parse_arg("data_choice")
+    min_wet_days = parse_arg("min_wet_days", int)
+    wet_day_def = parse_arg("wet_day_def", float)
+    lower_wet_threshold = parse_arg("lower_wet_threshold", int)
+    upper_wet_threshold = parse_arg("upper_wet_threshold", int)
+    maximum_temp = parse_arg("maximum_temp", float)
+    minimum_temp = parse_arg("minimum_temp", float)
+    temp_range = parse_arg("temp_range", float) 
+
+    x_min = pingrid.tile_left(tx, tz)
+    x_max = pingrid.tile_left(tx + 1, tz)
+
+    # row numbers increase as latitude decreases
+    y_max = pingrid.tile_top_mercator(ty, tz)
+    y_min = pingrid.tile_top_mercator(ty + 1, tz)
+    rr_mrg_year = rr_mrg.sel(T=rr_mrg['T.year']==target_year)
+    rr_mrg_season = rr_mrg_year.sel(T=rr_mrg_year["T.season"] == target_season)
+    tmin_mrg_year = tmin_mrg.sel(T=tmin_mrg['T.year']==target_year)
+    tmin_mrg_season = tmin_mrg_year.sel(T=tmin_mrg_year["T.season"] == target_season)
+    tmax_mrg_year = tmax_mrg.sel(T=tmax_mrg['T.year']==target_year)
+    tmax_mrg_season = tmax_mrg_year.sel(T=tmax_mrg_year["T.season"] == target_season)
+
+    if data_choice == "suitability_layer":
+        map_min = 0
+        map_max = 5
+        crop_suit_vals = crop_suitability(
+            rr_mrg_year, min_wet_days, wet_day_def, tmax_mrg_year, tmin_mrg_year, 
+            lower_wet_threshold, upper_wet_threshold, maximum_temp,
+            minimum_temp, temp_range, target_season 
+            ) 
+        data_tile = crop_suit_vals["crop_suit"]
+    else:
+        map_min = CONFIG["layers"][data_choice]["map_min"]
+        map_max = CONFIG["layers"][data_choice]["map_max"]
+        data_var = CONFIG["layers"][data_choice]["id"]
+        if data_choice == "precip_layer":
+            data_tile = rr_mrg_season
+        if data_choice == "tmin_layer":
+            data_tile = tmin_mrg_season
+        if data_choice == "tmax_layer":
+            data_tile = tmax_mrg_season
+    if (
+            # When we generalize this to other datasets, remember to
+            # account for the possibility that longitudes wrap around,
+            # so a < b doesn't always mean that a is west of b.
+            x_min > data_tile['X'].max() or
+            x_max < data_tile['X'].min() or
+            y_min > data_tile['Y'].max() or
+            y_max < data_tile['Y'].min()
+    ):
+        return pingrid.image_resp(pingrid.empty_tile())
```

Comment:
This short circuit should happen as early as possible, before we waste time trying to do the calculations.

Reply:
Check that what I did is appropriate